test/asm/string-formatting.asm

```diff
@@ -0,0 +1,9 @@
+n equ 300
+m equ -42
+f equ -123.0456
+s equs "hello"
+
+	printt "<{ -6d:n}> <{+06u:n}> <{5x:n}> <{#16b:n}>\n"
```

Comment:
Since spaces are significant, how about adding a test in cases where space trimming could break formats? (For example, macro args, but there may be other cases)